composite object init from operator expression

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"
    
type T struct {
    m uint16
}

var t = T{1<<2 | 1<<3}
    
func main() {
    fmt.Println(t)
}
```

Expected result:
```console
$ go run ./sample.go
{12}
```

Got:
```console
$ yaegi ./sample.go
panic: reflect.Value.Convert: value of type struct { Xm uint16 } cannot be converted to type uint16

goroutine 1 [running]:
reflect.Value.Convert(0xc0000ca8a0, 0xc0000c4aa8, 0x199, 0xfc50e0, 0xc9ece0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/reflect/value.go:2415 +0x224
github.com/containous/yaegi/interp.convertLiteralValue(0xc0000ef700, 0xfc50e0, 0xc9ece0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:2279 +0x99
github.com/containous/yaegi/interp.compositeLit(0xc0000ef500)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:1607 +0x178
github.com/containous/yaegi/interp.setExec.func1(0xc0000ef500)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1829 +0xb2
github.com/containous/yaegi/interp.setExec.func1(0xc0000ef700)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1818 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000efb00)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1818 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000ef800)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1818 +0x25d
github.com/containous/yaegi/interp.setExec(0xc0000ef800)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1832 +0xf5
github.com/containous/yaegi/interp.genRun.func1(0xc0000ef200, 0xc000197d48)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1559 +0x44
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ef200, 0xc000197d48, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:164 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ee500, 0xc000197d48, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.genRun(0xc0000ee500, 0xc0002264b0, 0xc0000c49f8)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1547 +0x64
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000238f20, 0xc0000c2280, 0x71, 0x271, 0xc0000c2280, 0x71, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:353 +0x388
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/composite5.go
+++ b/_test/composite5.go
@@ -1,0 +1,16 @@
+package main
+
+import "fmt"
+
+type T struct {
+	m uint16
+}
+
+var t = T{1<<2 | 1<<3}
+
+func main() {
+	fmt.Println(t)
+}
+
+// Output:
+// {12}

--- a/_test/composite6.go
+++ b/_test/composite6.go
@@ -1,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/containous/yaegi/_test/ct1"
+)
+
+type T struct {
+	m uint16
+}
+
+var t = T{1 << ct1.R}
+
+func main() {
+	fmt.Println(t)
+}
+
+// Output:
+// {2}

--- a/_test/ct1/ct1.go
+++ b/_test/ct1/ct1.go
@@ -1,0 +1,9 @@
+package ct1
+
+type Class uint
+
+const (
+	L Class = iota
+	R
+	AL
+)

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -231,7 +231,13 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 			}
 			// Propagate type to children, to handle implicit types
 			for _, c := range n.child {
-				c.typ = n.typ
+				switch c.kind {
+				case binaryExpr, unaryExpr:
+					// Do not attempt to propagate composite type to operator expressions,
+					// it breaks constant folding.
+				default:
+					c.typ = n.typ
+				}
 			}
 
 		case forStmt0, forRangeStmt:
@@ -1175,6 +1181,7 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 					n.gen = nop
 					n.typ = sym.typ
 					n.sym = sym
+					n.rval = sym.rval
 				} else {
 					err = n.cfgErrorf("undefined selector: %s.%s", pkg, name)
 				}

